releasenotes/notes/improve-link-content-fetcher-512d039e3c7684f1.yaml

```diff
@@ -0,0 +1,8 @@
+---
+enhancements:
+  - |
+    Several minor enhancements for LinkContentFetcher:
+    - Dynamic content handler resolution
+    - Custom User-Agent header (optional, minimize blocking)
+    - PDF support
+    - Register new content handlers
```

Comment:
For the next time: if you can make a list of features you're adding to a component, that's a missed opportunity to make a list of PRs, one for each improvement. Small PRs go through much quicker!

Reply:
Ok, they were all kinda intertwined and small, but I see your point. I will do so next time.